components/backup-stream/src/utils.rs

```diff
@@ -589,6 +594,42 @@ pub fn is_overlapping(range: (&[u8], &[u8]), range2: (&[u8], &[u8])) -> bool {
     }
 }
 
+pub struct FilesReader {
+    files: Vec<Pin<Box<File>>>,
```

Comment:
I think we don't need `Pin` here because `File` is naturally `Unpin`. BTW, we can make it generic over `Files` by constraint it with `AsyncRead`, or why not trivially use [`AsyncReadExt::chain`](https://docs.rs/tokio/latest/tokio/io/trait.AsyncReadExt.html#method.chain)?